### PR TITLE
Booking step2 guest

### LINKED ISSUE #19: Booking step 2 - GUI  and logic for continue as guest
- GUI for booking step 2 as guest
- Logic to pass data to next booking step

Design: https://www.figma.com/file/bRPboJBlptgd24c9WWHxEH/UI%2FUX---Gruppuppgift-2?type=design&node-id=0-1&t=pKdX6fVnLQtNDe99-0

### PR DESCRIPTION
**Pull request by mkor92**
Booking step 2 as guest. 
- Input form with validation.
- Ticketcount from booking step 1 passed as props.
- As for now it saves data to object.

Screenshot 2023-05-09 095700

Screenshot 2023-05-03 151410

### CODE REVIEW

**Review by matodorovic — APPROVED** [review]
👍

**Review by jakobpalmborg — CHANGES_REQUESTED** [review]
Looking good!  Great work ! I left a few comments. Maybe we should think about those things before we merge this PR.

File: src/app/(site)/tickets/page.tsx [hunk 1]
```diff
@@ -1,294 +1,85 @@
...
+  const handleGuest = () => {
+    setContinueGuest(true);
+  };
 
-      <div className="total-sum-container">
-        <h2>Att betala:</h2>
-        <h2>{ticketCount * 150}kr</h2>
-      </div>
+  if (continueGuest) {
+    return <GuestTickets tickets={ticketCount} />;
```
Comment (jakobpalmborg) [inline src/app/(site)/tickets/page.tsx]: What do you need ticketCount for here? I think we need to create state for the booking with screeningId, seatNumber and email instead. That is what we need to create a ticket.

File: src/app/components/GuestTickets.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    tickets: string;
+  }>({
+    name: '',
+    email: '',
+    number: '',
+    tickets: `${props.tickets}`,
+  });
+  const inputs = [
+    {
+      id: 1,
```
Comment (jakobpalmborg) [inline src/app/components/GuestTickets.tsx:16]: In the design it looks like this but we do not have anywhere in the database to store name and phone for guest users as it is right now. We are just storing the email and a think that that could be enough. We could skip the phone number, send the email with the ticket booking and use the name as a greeting in the end. For example: "Thank you, Markus". And not save the name in the database. What do you think about that?

File: src/app/components/GuestTickets.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
+'use client';
+import { useState } from 'react';
+import FormInput from '@/app/components/FormInput';
+import Link from 'next/link';
+
+export default function signup(props: any) {
+  const [guestData, setGuestData] = useState<{
```
Comment (jakobpalmborg) [inline src/app/components/GuestTickets.tsx:7]: how will we use this i the rest of the application. See my other comment about what we will save to the database.

File: src/app/(site)/tickets/page.tsx (hunk 1 above)
Comment (mkor92) [reply src/app/(site)/tickets/page.tsx]: I think we need to pass ticketCount to payment, so customer can se total price at the end? "Totalt att betala: "

File: src/app/components/GuestTickets.tsx (hunk 2 above)
Comment (mkor92) [reply src/app/components/GuestTickets.tsx:16]: I followed the design. But i guess for now we dont need the number. What do you mean about not saving name? If we are going to show name + greeting in the end. We need to save the name somehow?

File: src/app/components/GuestTickets.tsx (hunk 3 above)
Comment (mkor92) [reply src/app/components/GuestTickets.tsx:7]: Logic for going from booking step 2 - 3 is not done. Im not sure yet, just an example code. But i think its a good idea to save data to object. And in the end we can choose which data is going to be in the POST request for creating a ticket and which data we only use to the "Thanks for booking message"

File: src/app/(site)/tickets/page.tsx (hunk 1 above)
Comment (jakobpalmborg) [reply src/app/(site)/tickets/page.tsx]: With React we can save it in state during the ticket booking.

File: src/app/components/GuestTickets.tsx (hunk 2 above)
Comment (jakobpalmborg) [reply src/app/components/GuestTickets.tsx:16]: As before, we can use React for that.

File: src/app/components/GuestTickets.tsx (hunk 3 above)
Comment (jakobpalmborg) [reply src/app/components/GuestTickets.tsx:7]: The whole booking process is dynamically rendered so I don't think there is needed much logic to go to next step. Just update state and conditionally render the right things.

**Review by jakobpalmborg — APPROVED** [review]

**Pull request merged**